src/mount.rs

```diff
@@ -49,6 +49,13 @@ pub fn mount_root(options: &CmdlineOptions) -> Result<()> {
         return Err("root= not found in /proc/cmdline".into());
     }
 
+    if options.rootfstype != Some("nfs".to_string()) &&
+        options.rootfstype != Some("9p".to_string())
+    {
+        let root_device = options.root.as_ref().ok_or("No root device argument")?;
+        wait_for_device(root_device)?;
+    }
+
```

Comment:
```suggestion
    match options.rootfstype.as_deref() {
        Some("nfs") | Some("9p") => (),
        _ => wait_for_device(root)?,
    }
```

This relies on a change I made above which I can not format as suggestion here. My local `mount_root` looks like this:

```rust
pub fn mount_root(options: &CmdlineOptions) -> Result<()> {
    let root = options
        .root
        .as_ref()
        .ok_or("root= not found in /proc/cmdline")?;

    match options.rootfstype.as_deref() {
        Some("nfs") | Some("9p") => (),
        _ => wait_for_device(root)?,
    }

    mkdir("/root")?;

    debug!(
        "Mounting rootfs {} -> /root as {} with flags = {:#x}, data = '{}'",
        root,
        options.rootfstype.as_deref().unwrap_or_default(),
        options.rootfsflags.bits(),
        options.rootflags.as_deref().unwrap_or_default()
    );
    do_mount(
        Some(root),
        "/root",
        options.rootfstype.as_deref(),
        options.rootfsflags,
        options.rootflags.as_deref(),
    )?;

    Ok(())
}
```